plugin/ochttp/server.go

```diff
@@ -140,8 +140,12 @@ type trackingResponseWriter struct {
 	writer     http.ResponseWriter
 }
 
-var _ http.ResponseWriter = (*trackingResponseWriter)(nil)
+// Compile time assertions for widely used net/http interfaces
+var _ http.CloseNotifier = (*trackingResponseWriter)(nil)
+var _ http.Flusher = (*trackingResponseWriter)(nil)
 var _ http.Hijacker = (*trackingResponseWriter)(nil)
+var _ http.Pusher = (*trackingResponseWriter)(nil)
+var _ http.ResponseWriter = (*trackingResponseWriter)(nil)
```

Comment:
what about ReaderFrom as mentioned in the linked issue? is that intentionally left to a future PR? Also probably should implement WriterTo. These are more involved (they need to work even if the underlying streams don't support them directly) so maybe we need a separate issue tracking them.